<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>CrossroadXb</class>
 <widget class="QWidget" name="CrossroadXb">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1160</width>
    <height>520</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>1160</width>
    <height>520</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>1160</width>
    <height>520</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <property name="styleSheet">
   <string notr="true">background-color: rgb(186, 189, 182);</string>
  </property>
  <widget class="Line" name="separationLine_8">
   <property name="geometry">
    <rect>
     <x>274</x>
     <y>18</y>
     <width>8</width>
     <height>145</height>
    </rect>
   </property>
   <property name="frameShadow">
    <enum>QFrame::Plain</enum>
   </property>
   <property name="lineWidth">
    <number>3</number>
   </property>
   <property name="orientation">
    <enum>Qt::Vertical</enum>
   </property>
  </widget>
  <widget class="Line" name="separationLine_7">
   <property name="geometry">
    <rect>
     <x>174</x>
     <y>18</y>
     <width>8</width>
     <height>145</height>
    </rect>
   </property>
   <property name="frameShadow">
    <enum>QFrame::Plain</enum>
   </property>
   <property name="lineWidth">
    <number>3</number>
   </property>
   <property name="orientation">
    <enum>Qt::Vertical</enum>
   </property>
  </widget>
  <widget class="QLabel" name="label_2">
   <property name="geometry">
    <rect>
     <x>574</x>
     <y>3</y>
     <width>181</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600; font-style:italic;&quot;&gt;Conventional semaphore&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;br/&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
   </property>
  </widget>
  <widget class="QPushButton" name="buttonC2">
   <property name="geometry">
    <rect>
     <x>244</x>
     <y>3</y>
     <width>21</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>S</string>
   </property>
  </widget>
  <widget class="QPushButton" name="buttonC6">
   <property name="geometry">
    <rect>
     <x>4</x>
     <y>310</y>
     <width>21</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>J</string>
   </property>
  </widget>
  <widget class="Line" name="separationLine_14">
   <property name="geometry">
    <rect>
     <x>954</x>
     <y>365</y>
     <width>8</width>
     <height>145</height>
    </rect>
   </property>
   <property name="frameShadow">
    <enum>QFrame::Plain</enum>
   </property>
   <property name="lineWidth">
    <number>3</number>
   </property>
   <property name="orientation">
    <enum>Qt::Vertical</enum>
   </property>
  </widget>
  <widget class="QLabel" name="label_3">
   <property name="geometry">
    <rect>
     <x>384</x>
     <y>83</y>
     <width>93</width>
     <height>17</height>
    </rect>
   </property>
   <property name="text">
    <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600; font-style:italic;&quot;&gt;Cars passed: &lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;br/&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
   </property>
  </widget>
  <widget class="Line" name="separationLine_21">
   <property name="geometry">
    <rect>
     <x>610</x>
     <y>161</y>
     <width>145</width>
     <height>8</height>
    </rect>
   </property>
   <property name="frameShadow">
    <enum>QFrame::Plain</enum>
   </property>
   <property name="lineWidth">
    <number>3</number>
   </property>
   <property name="orientation">
    <enum>Qt::Horizontal</enum>
   </property>
  </widget>
  <widget class="QPushButton" name="buttonC3">
   <property name="geometry">
    <rect>
     <x>534</x>
     <y>210</y>
     <width>21</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>K</string>
   </property>
  </widget>
  <widget class="Line" name="separationLine_11">
   <property name="geometry">
    <rect>
     <x>274</x>
     <y>363</y>
     <width>8</width>
     <height>145</height>
    </rect>
   </property>
   <property name="frameShadow">
    <enum>QFrame::Plain</enum>
   </property>
   <property name="lineWidth">
    <number>3</number>
   </property>
   <property name="orientation">
    <enum>Qt::Vertical</enum>
   </property>
  </widget>
  <widget class="Semaphore" name="semS_1_2">
   <property name="geometry">
    <rect>
     <x>384</x>
     <y>363</y>
     <width>16</width>
     <height>31</height>
    </rect>
   </property>
   <property name="frameShape">
    <enum>QFrame::StyledPanel</enum>
   </property>
   <property name="frameShadow">
    <enum>QFrame::Raised</enum>
   </property>
  </widget>
  <widget class="Line" name="separationLine_27">
   <property name="geometry">
    <rect>
     <x>806</x>
     <y>20</y>
     <width>5</width>
     <height>145</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true">color: rgb(255, 255, 255);</string>
   </property>
   <property name="frameShadow">
    <enum>QFrame::Plain</enum>
   </property>
   <property name="lineWidth">
    <number>3</number>
   </property>
   <property name="orientation">
    <enum>Qt::Vertical</enum>
   </property>
  </widget>
  <widget class="QLabel" name="label_4">
   <property name="geometry">
    <rect>
     <x>972</x>
     <y>93</y>
     <width>93</width>
     <height>17</height>
    </rect>
   </property>
   <property name="text">
    <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600; font-style:italic;&quot;&gt;Cars passed: &lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;br/&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
   </property>
  </widget>
  <widget class="Line" name="separationLine_9">
   <property name="geometry">
    <rect>
     <x>374</x>
     <y>18</y>
     <width>8</width>
     <height>145</height>
    </rect>
   </property>
   <property name="frameShadow">
    <enum>QFrame::Plain</enum>
   </property>
   <property name="lineWidth">
    <number>3</number>
   </property>
   <property name="orientation">
    <enum>Qt::Vertical</enum>
   </property>
  </widget>
  <widget class="Semaphore" name="semS_2_2">
   <property name="geometry">
    <rect>
     <x>254</x>
     <y>363</y>
     <width>16</width>
     <height>31</height>
    </rect>
   </property>
   <property name="frameShape">
    <enum>QFrame::StyledPanel</enum>
   </property>
   <property name="frameShadow">
    <enum>QFrame::Raised</enum>
   </property>
  </widget>
  <widget class="Line" name="separationLine_19">
   <property name="geometry">
    <rect>
     <x>960</x>
     <y>359</y>
     <width>145</width>
     <height>8</height>
    </rect>
   </property>
   <property name="frameShadow">
    <enum>QFrame::Plain</enum>
   </property>
   <property name="lineWidth">
    <number>3</number>
   </property>
   <property name="orientation">
    <enum>Qt::Horizontal</enum>
   </property>
  </widget>
  <widget class="QLineEdit" name="totalCarsConv">
   <property name="geometry">
    <rect>
     <x>1064</x>
     <y>93</y>
     <width>51</width>
     <height>24</height>
    </rect>
   </property>
  </widget>
  <widget class="Semaphore" name="semC_1_2">
   <property name="geometry">
    <rect>
     <x>964</x>
     <y>367</y>
     <width>16</width>
     <height>31</height>
    </rect>
   </property>
   <property name="frameShape">
    <enum>QFrame::StyledPanel</enum>
   </property>
   <property name="frameShadow">
    <enum>QFrame::Raised</enum>
   </property>
  </widget>
  <widget class="Semaphore" name="semC_2_1">
   <property name="geometry">
    <rect>
     <x>864</x>
     <y>133</y>
     <width>16</width>
     <height>31</height>
    </rect>
   </property>
   <property name="frameShape">
    <enum>QFrame::StyledPanel</enum>
   </property>
   <property name="frameShadow">
    <enum>QFrame::Raised</enum>
   </property>
  </widget>
  <widget class="Line" name="separationMiddle">
   <property name="geometry">
    <rect>
     <x>564</x>
     <y>1</y>
     <width>10</width>
     <height>515</height>
    </rect>
   </property>
   <property name="frameShadow">
    <enum>QFrame::Plain</enum>
   </property>
   <property name="lineWidth">
    <number>3</number>
   </property>
   <property name="orientation">
    <enum>Qt::Vertical</enum>
   </property>
  </widget>
  <widget class="Line" name="separationLine_26">
   <property name="geometry">
    <rect>
     <x>326</x>
     <y>363</y>
     <width>5</width>
     <height>145</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true">color: rgb(255, 255, 255);</string>
   </property>
   <property name="frameShadow">
    <enum>QFrame::Plain</enum>
   </property>
   <property name="lineWidth">
    <number>3</number>
   </property>
   <property name="orientation">
    <enum>Qt::Vertical</enum>
   </property>
  </widget>
  <widget class="QPushButton" name="buttonC5">
   <property name="geometry">
    <rect>
     <x>294</x>
     <y>503</y>
     <width>21</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>D</string>
   </property>
  </widget>
  <widget class="Line" name="separationLine_16">
   <property name="geometry">
    <rect>
     <x>610</x>
     <y>260</y>
     <width>145</width>
     <height>8</height>
    </rect>
   </property>
   <property name="frameShadow">
    <enum>QFrame::Plain</enum>
   </property>
   <property name="lineWidth">
    <number>3</number>
   </property>
   <property name="orientation">
    <enum>Qt::Horizontal</enum>
   </property>
  </widget>
  <widget class="Line" name="separationLine_20">
   <property name="geometry">
    <rect>
     <x>954</x>
     <y>20</y>
     <width>8</width>
     <height>145</height>
    </rect>
   </property>
   <property name="frameShadow">
    <enum>QFrame::Plain</enum>
   </property>
   <property name="lineWidth">
    <number>3</number>
   </property>
   <property name="orientation">
    <enum>Qt::Vertical</enum>
   </property>
  </widget>
  <widget class="Line" name="separationLine_12">
   <property name="geometry">
    <rect>
     <x>374</x>
     <y>363</y>
     <width>8</width>
     <height>145</height>
    </rect>
   </property>
   <property name="frameShadow">
    <enum>QFrame::Plain</enum>
   </property>
   <property name="lineWidth">
    <number>3</number>
   </property>
   <property name="orientation">
    <enum>Qt::Vertical</enum>
   </property>
  </widget>
  <widget class="Line" name="separationLine_3">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>358</y>
     <width>145</width>
     <height>8</height>
    </rect>
   </property>
   <property name="frameShadow">
    <enum>QFrame::Plain</enum>
   </property>
   <property name="lineWidth">
    <number>3</number>
   </property>
   <property name="orientation">
    <enum>Qt::Horizontal</enum>
   </property>
  </widget>
  <widget class="QLineEdit" name="totalCarsSmart">
   <property name="geometry">
    <rect>
     <x>476</x>
     <y>83</y>
     <width>51</width>
     <height>24</height>
    </rect>
   </property>
   <property name="frame">
    <bool>true</bool>
   </property>
  </widget>
  <widget class="Line" name="separationLine_6">
   <property name="geometry">
    <rect>
     <x>380</x>
     <y>158</y>
     <width>145</width>
     <height>8</height>
    </rect>
   </property>
   <property name="frameShadow">
    <enum>QFrame::Plain</enum>
   </property>
   <property name="lineWidth">
    <number>3</number>
   </property>
   <property name="orientation">
    <enum>Qt::Horizontal</enum>
   </property>
  </widget>
  <widget class="Semaphore" name="semS_3_1">
   <property name="geometry">
    <rect>
     <x>144</x>
     <y>366</y>
     <width>31</width>
     <height>16</height>
    </rect>
   </property>
   <property name="frameShape">
    <enum>QFrame::StyledPanel</enum>
   </property>
   <property name="frameShadow">
    <enum>QFrame::Raised</enum>
   </property>
  </widget>
  <widget class="Line" name="separationLine_10">
   <property name="geometry">
    <rect>
     <x>174</x>
     <y>363</y>
     <width>8</width>
     <height>145</height>
    </rect>
   </property>
   <property name="frameShadow">
    <enum>QFrame::Plain</enum>
   </property>
   <property name="lineWidth">
    <number>3</number>
   </property>
   <property name="orientation">
    <enum>Qt::Vertical</enum>
   </property>
  </widget>
  <widget class="QLabel" name="label">
   <property name="geometry">
    <rect>
     <x>434</x>
     <y>3</y>
     <width>131</width>
     <height>17</height>
    </rect>
   </property>
   <property name="text">
    <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600; font-style:italic;&quot;&gt;Smart semaphore &lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;br/&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
   </property>
  </widget>
  <widget class="Line" name="separationLine_17">
   <property name="geometry">
    <rect>
     <x>854</x>
     <y>20</y>
     <width>8</width>
     <height>145</height>
    </rect>
   </property>
   <property name="frameShadow">
    <enum>QFrame::Plain</enum>
   </property>
   <property name="lineWidth">
    <number>3</number>
   </property>
   <property name="orientation">
    <enum>Qt::Vertical</enum>
   </property>
  </widget>
  <widget class="Semaphore" name="semS_4_1">
   <property name="geometry">
    <rect>
     <x>380</x>
     <y>140</y>
     <width>31</width>
     <height>16</height>
    </rect>
   </property>
   <property name="frameShape">
    <enum>QFrame::StyledPanel</enum>
   </property>
   <property name="frameShadow">
    <enum>QFrame::Raised</enum>
   </property>
  </widget>
  <widget class="QPushButton" name="buttonC1">
   <property name="geometry">
    <rect>
     <x>194</x>
     <y>3</y>
     <width>21</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>A</string>
   </property>
  </widget>
  <widget class="Line" name="separationLine_18">
   <property name="geometry">
    <rect>
     <x>754</x>
     <y>362</y>
     <width>8</width>
     <height>145</height>
    </rect>
   </property>
   <property name="frameShadow">
    <enum>QFrame::Plain</enum>
   </property>
   <property name="lineWidth">
    <number>3</number>
   </property>
   <property name="orientation">
    <enum>Qt::Vertical</enum>
   </property>
  </widget>
  <widget class="Line" name="separationLine_2">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>258</y>
     <width>145</width>
     <height>8</height>
    </rect>
   </property>
   <property name="frameShadow">
    <enum>QFrame::Plain</enum>
   </property>
   <property name="lineWidth">
    <number>3</number>
   </property>
   <property name="orientation">
    <enum>Qt::Horizontal</enum>
   </property>
  </widget>
  <widget class="Line" name="separationLine_22">
   <property name="geometry">
    <rect>
     <x>854</x>
     <y>365</y>
     <width>8</width>
     <height>145</height>
    </rect>
   </property>
   <property name="frameShadow">
    <enum>QFrame::Plain</enum>
   </property>
   <property name="lineWidth">
    <number>3</number>
   </property>
   <property name="orientation">
    <enum>Qt::Vertical</enum>
   </property>
  </widget>
  <widget class="Semaphore" name="semC_3_1">
   <property name="geometry">
    <rect>
     <x>724</x>
     <y>368</y>
     <width>31</width>
     <height>16</height>
    </rect>
   </property>
   <property name="frameShape">
    <enum>QFrame::StyledPanel</enum>
   </property>
   <property name="frameShadow">
    <enum>QFrame::Raised</enum>
   </property>
  </widget>
  <widget class="Line" name="separationLine_13">
   <property name="geometry">
    <rect>
     <x>754</x>
     <y>20</y>
     <width>8</width>
     <height>145</height>
    </rect>
   </property>
   <property name="frameShadow">
    <enum>QFrame::Plain</enum>
   </property>
   <property name="lineWidth">
    <number>3</number>
   </property>
   <property name="orientation">
    <enum>Qt::Vertical</enum>
   </property>
  </widget>
  <widget class="Semaphore" name="semS_1_1">
   <property name="geometry">
    <rect>
     <x>154</x>
     <y>128</y>
     <width>16</width>
     <height>31</height>
    </rect>
   </property>
   <property name="frameShape">
    <enum>QFrame::StyledPanel</enum>
   </property>
   <property name="frameShadow">
    <enum>QFrame::Raised</enum>
   </property>
  </widget>
  <widget class="Line" name="separationLine_1">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>159</y>
     <width>145</width>
     <height>8</height>
    </rect>
   </property>
   <property name="frameShadow">
    <enum>QFrame::Plain</enum>
   </property>
   <property name="lineWidth">
    <number>3</number>
   </property>
   <property name="orientation">
    <enum>Qt::Horizontal</enum>
   </property>
  </widget>
  <widget class="Semaphore" name="semC_2_2">
   <property name="geometry">
    <rect>
     <x>834</x>
     <y>363</y>
     <width>16</width>
     <height>31</height>
    </rect>
   </property>
   <property name="frameShape">
    <enum>QFrame::StyledPanel</enum>
   </property>
   <property name="frameShadow">
    <enum>QFrame::Raised</enum>
   </property>
  </widget>
  <widget class="QPushButton" name="buttonC4">
   <property name="geometry">
    <rect>
     <x>344</x>
     <y>503</y>
     <width>21</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>F</string>
   </property>
  </widget>
  <widget class="Line" name="separationLine_28">
   <property name="geometry">
    <rect>
     <x>906</x>
     <y>366</y>
     <width>5</width>
     <height>145</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true">color: rgb(255, 255, 255);</string>
   </property>
   <property name="frameShadow">
    <enum>QFrame::Plain</enum>
   </property>
   <property name="lineWidth">
    <number>3</number>
   </property>
   <property name="orientation">
    <enum>Qt::Vertical</enum>
   </property>
  </widget>
  <widget class="Line" name="separationLine_23">
   <property name="geometry">
    <rect>
     <x>960</x>
     <y>259</y>
     <width>145</width>
     <height>8</height>
    </rect>
   </property>
   <property name="frameShadow">
    <enum>QFrame::Plain</enum>
   </property>
   <property name="lineWidth">
    <number>3</number>
   </property>
   <property name="orientation">
    <enum>Qt::Horizontal</enum>
   </property>
  </widget>
  <widget class="Line" name="separationLine_5">
   <property name="geometry">
    <rect>
     <x>380</x>
     <y>257</y>
     <width>145</width>
     <height>8</height>
    </rect>
   </property>
   <property name="frameShadow">
    <enum>QFrame::Plain</enum>
   </property>
   <property name="lineWidth">
    <number>3</number>
   </property>
   <property name="orientation">
    <enum>Qt::Horizontal</enum>
   </property>
  </widget>
  <widget class="Semaphore" name="semC_1_1">
   <property name="geometry">
    <rect>
     <x>734</x>
     <y>133</y>
     <width>16</width>
     <height>31</height>
    </rect>
   </property>
   <property name="frameShape">
    <enum>QFrame::StyledPanel</enum>
   </property>
   <property name="frameShadow">
    <enum>QFrame::Raised</enum>
   </property>
  </widget>
  <widget class="QPushButton" name="startSimulation">
   <property name="geometry">
    <rect>
     <x>4</x>
     <y>13</y>
     <width>131</width>
     <height>25</height>
    </rect>
   </property>
   <property name="text">
    <string>Start simulation</string>
   </property>
  </widget>
  <widget class="Line" name="separationLine_25">
   <property name="geometry">
    <rect>
     <x>226</x>
     <y>18</y>
     <width>5</width>
     <height>145</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true">color: rgb(255, 255, 255);</string>
   </property>
   <property name="frameShadow">
    <enum>QFrame::Plain</enum>
   </property>
   <property name="lineWidth">
    <number>3</number>
   </property>
   <property name="orientation">
    <enum>Qt::Vertical</enum>
   </property>
  </widget>
  <widget class="Line" name="separationLine_4">
   <property name="geometry">
    <rect>
     <x>380</x>
     <y>357</y>
     <width>145</width>
     <height>8</height>
    </rect>
   </property>
   <property name="frameShadow">
    <enum>QFrame::Plain</enum>
   </property>
   <property name="lineWidth">
    <number>3</number>
   </property>
   <property name="orientation">
    <enum>Qt::Horizontal</enum>
   </property>
  </widget>
  <widget class="Semaphore" name="semS_2_1">
   <property name="geometry">
    <rect>
     <x>284</x>
     <y>133</y>
     <width>16</width>
     <height>31</height>
    </rect>
   </property>
   <property name="frameShape">
    <enum>QFrame::StyledPanel</enum>
   </property>
   <property name="frameShadow">
    <enum>QFrame::Raised</enum>
   </property>
  </widget>
  <widget class="Line" name="separationLine_24">
   <property name="geometry">
    <rect>
     <x>960</x>
     <y>160</y>
     <width>145</width>
     <height>8</height>
    </rect>
   </property>
   <property name="frameShadow">
    <enum>QFrame::Plain</enum>
   </property>
   <property name="lineWidth">
    <number>3</number>
   </property>
   <property name="orientation">
    <enum>Qt::Horizontal</enum>
   </property>
  </widget>
  <widget class="Semaphore" name="semC_4_1">
   <property name="geometry">
    <rect>
     <x>960</x>
     <y>140</y>
     <width>31</width>
     <height>16</height>
    </rect>
   </property>
   <property name="frameShape">
    <enum>QFrame::StyledPanel</enum>
   </property>
   <property name="frameShadow">
    <enum>QFrame::Raised</enum>
   </property>
  </widget>
  <widget class="Line" name="separationLine_15">
   <property name="geometry">
    <rect>
     <x>610</x>
     <y>360</y>
     <width>145</width>
     <height>8</height>
    </rect>
   </property>
   <property name="frameShadow">
    <enum>QFrame::Plain</enum>
   </property>
   <property name="lineWidth">
    <number>3</number>
   </property>
   <property name="orientation">
    <enum>Qt::Horizontal</enum>
   </property>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>Semaphore</class>
   <extends>QFrame</extends>
   <header>semaphore.hpp</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
